app/presenters/vendor_api/single_application_presenter.rb

```diff
@@ -260,5 +264,9 @@ def work_history_breaks
         ''
       end
     end
+
+    def safeguarding_issues_details_url
+      application_form.has_safeguarding_issues_to_declare? ? provider_interface_application_choice_url(application_choice).concat('#criminal-convictions-and-professional-misconduct') : nil
```

Comment:
I think Rails path helpers support an `anchor:` property (https://stackoverflow.com/questions/51895357/rails-5-appending-an-html-anchor-to-rails-route-path-helper)